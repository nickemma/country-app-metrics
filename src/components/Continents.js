import { useDispatch, useSelector } from 'react-redux';
import { BsFillArrowRightSquareFill } from 'react-icons/bs';
import { useParams } from 'react-router';
import continents from '../data/data';
import { getCountries } from '../redux/countries/countries';
// import '../styles/continents.scss';

const Continents = () => {
  const { idName } = useParams();
  const dispatch = useDispatch();
  const countriesOfContinent = useSelector((state) => state.countries);
  console.log('fdfd', countriesOfContinent);
  const handleClick = (name) => {
    dispatch(getCountries(name, idName));
  };

  return (
    <div>
      <ul className="continents">
        {continents.map(({ id, name, image }) => (
          <li key={id}>
            {name}
            <img src={image} alt="awesome" />
            <button type="button" onClick={() => handleClick(name)}>
              <BsFillArrowRightSquareFill />
            </button>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Continents;
