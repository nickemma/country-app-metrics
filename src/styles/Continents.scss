.continents {
  .items {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(32rem, 1fr));
    gap: 2rem;

    li {
      display: flex;
      align-items: center;
      justify-content: center;

      img {
        width: 100%;
        height: 100%;
        object-fit: cover;
      }

      h2 {
        font-size: 2.5rem;
        margin-top: 30px;
      }

      a {
        font-size: 2rem;
        color: var(--text-color);
        font-weight: 600;
        animation: floating 2s linear infinite;

        @keyframes floating {
          0% {
            transform: translateY(0);
          }

          100% {
            transform: translateY(0);
          }

          50% {
            transform: translateY(-1rem);
          }
        }
      }
    }

    li:nth-child(even) {
      border-radius: 0.5rem;
      padding: 5rem;
      margin: 10px;
    }

    li:nth-child(odd) {
      background: var(--primary-color);
      border-radius: 0.5rem;
      padding: 5rem;
      margin: 10px;
    }
  }
}
