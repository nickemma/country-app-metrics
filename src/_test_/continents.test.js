import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Continents from '../components/Continents';

it('renders correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <Continents />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
