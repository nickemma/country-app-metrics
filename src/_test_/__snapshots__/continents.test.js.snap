// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="continents"
>
  <ul
    className="items"
  >
    <li>
      <img
        alt="awesome"
        src="/images/Africa.svg"
      />
      <h2>
        Africa
      </h2>
      <a
        href="/Africa"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </a>
    </li>
    <li>
      <img
        alt="awesome"
        src="/images/Asia.svg"
      />
      <h2>
        Asia
      </h2>
      <a
        href="/Asia"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </a>
    </li>
    <li>
      <img
        alt="awesome"
        src="/images/Europe.svg"
      />
      <h2>
        Europe
      </h2>
      <a
        href="/Europe"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </a>
    </li>
    <li>
      <img
        alt="awesome"
        src="/images/Oceania.svg"
      />
      <h2>
        Oceania
      </h2>
      <a
        href="/Oceania"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </a>
    </li>
    <li>
      <img
        alt="awesome"
        src="/images/North America.svg"
      />
      <h2>
        North America
      </h2>
      <a
        href="/North America"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </a>
    </li>
    <li>
      <img
        alt="awesome"
        src="/images/South America.svg"
      />
      <h2>
        South America
      </h2>
      <a
        href="/South America"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </a>
    </li>
    <li>
      <img
        alt="awesome"
        src="/images/Antarctica.svg"
      />
      <h2>
        Antarctica
      </h2>
      <a
        href="/Antarctica"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </a>
    </li>
  </ul>
</section>
`;
